test/framework/src/main/java/org/elasticsearch/ingest/TestIngestDocument.java

```diff
@@ -34,10 +39,30 @@ public static IngestDocument ofSourceAndMetadata(Map<String, Object> sourceAndMe
         return new IngestDocument(sourceAndMetadata, new HashMap<>());
     }
 
+    /**
+     * Create an IngestDocument for testing but takes a pre-constructed {@link IngestSourceAndMetadata}
```

Comment:
this comment doesn't seem to match the signature

Reply:
Fixed.